### ISSUE TITLE
Security confidence addition to the images for more clear traceability and security

### ISSUE THREAD

**Issue by harshad16** [thoth-station/thoth-application#1813]
**Is your feature request related to a problem? Please describe.**

The plan is to have a process in place to add score/confidence based on the base image and package security into the image during the build time. we would focus on the provenance of the package and the base image source confidence. The aggregation would be provided in the image for the details for the user to view and verify that the built image is of good quality.

This is not the same as the vulnerability of the installed binary, but it would be similar to it in the context of the base image and packages installed. 
The goal is have a process:
[ ] phase 1. To accumulate the score/confidence based on the base image and package (for example: thamos advise security score, base image source confidence based on if it is official or not)
[ ] phase 2. To determine the method to include them in a format into the image. 
[ ] phase 3 . Introduce an open policy to check back on what this information signifies. 

References
opa: open policy agent 
sigstore: signature for the package https://sigstore.dev/what_is_sigstore/
stackrox: https://www.stackrox.com/use-cases/vulnerability-management/

**Comment by goern** [comment]
is this still valid @pacospace

**Comment by pacospace** [comment]
> is this still valid @pacospace

Yes, we do not collect scorecards yet: https://github.com/thoth-station/core/issues/289

**Comment by fridex** [comment]
Looking into this issue, it might be better to have security information on the backend side linked to the container image SHA (and let clients ask for it). If you take a look at how the vulnerabilities or issues are discovered - they are discovered not during the build time but rather as the container image is used. A solution that would link container image SHA to known issues or vulnerabilities would scale long term and would provide the flexibility to add observations *after* the container image is built based on its use (ex. exposed as an API endpoint). Moreover, we could also aggregate queries stats (that can provide insights on the container image use by users).
